### PR TITLE
Add support for MsSQL

### PR DESCRIPTION
**Pull request by TheConstructor**
I ported the MsSqlContainer to .NET, because I (and presumably others) may want to use it.

Tested running on Mac OS X

### CODE REVIEW

File: TestContainers/Core/Containers/DatabaseContainer.cs [hunk 1]
```diff
@@ -2,9 +2,7 @@ namespace TestContainers.Core.Containers
 {
     public abstract class DatabaseContainer : Container
     {
-        protected int GetStartupTimeoutSeconds => 120;
-
-        protected int GetConnectTImeoutSeconds => 120;
+        protected virtual int GetConnectTimeoutSeconds => 120;
```
Comment (swissarmykirpan) [inline TestContainers/Core/Containers/DatabaseContainer.cs]: might be worth turning this into a `TimeSpan` to give it more flexibility

File: TestContainers/Core/Containers/MsSqlContainer.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+namespace TestContainers.Core.Containers
+{
+    public sealed class MsSqlContainer : DatabaseContainer
+    {
+        public const string NAME = "mssqlserver";
+        public const string IMAGE = "mcr.microsoft.com/mssql/server";
+        public const string DEFAULT_TAG = "2017-latest-ubuntu";
+        
+        /// <summary>This will run the container using the Developer Edition (this is the default if no MSSQL_PID environment variable is supplied)</summary>
```
Comment (swissarmykirpan) [inline TestContainers/Core/Containers/MsSqlContainer.cs]: can we remove the comments here please

**Review by swissarmykirpan — CHANGES_REQUESTED** [review]
small feedback

**Comment by TheConstructor** [comment]
@swissarmykirpan I hope I did everything the way you wanted :)

File: TestContainers.Tests/ContainerTests/MsSqlTests.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+namespace TestContainers.Tests.ContainerTests
+{
+    public class MsSqlFixture : IAsyncLifetime
+    {
+        public string ConnectionString => Container.ConnectionString;
+        MsSqlContainer Container { get; }
+
+        public MsSqlFixture()
+        {
+            LicenseAcceptance.AcceptLicense("mcr.microsoft.com/mssql/server");
```
Comment (SbiCA) [inline TestContainers.Tests/ContainerTests/MsSqlTests.cs:16]: As a user of test container this seems strange to me, Why would I need to call another static calls for accepting the licencse? I would expect that I can use the Fluent API on the `DatabaseContainerBuilder` with something like `WithAcceptEula()`

File: TestContainers/Core/Containers/MsSqlContainer.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+namespace TestContainers.Core.Containers
+{
+    public sealed class MsSqlContainer : DatabaseContainer
+    {
+        public const string NAME = "mssqlserver";
+        public const string IMAGE = "mcr.microsoft.com/mssql/server";
+        public const string DEFAULT_TAG = "2017-latest-ubuntu";
+
+        public const string EDITION_DEVELOPER = "Developer";
```
Comment (SbiCA) [inline TestContainers/Core/Containers/MsSqlContainer.cs:15]: Maybe use a typed enum for the editions https://docs.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/enumeration-classes-over-enum-types

File: TestContainers/Core/Containers/MsSqlContainer.cs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        public const string EDITION_EXPRESS = "Express";
+        public const string EDITION_STANDARD = "Standard";
+        public const string EDITION_ENTERPRISE = "Enterprise";
+        public const string EDITION_ENTERPRISECORE = "EnterpriseCore";
+
+        public const int MSSQL_PORT = 1433;
+
+        protected override TimeSpan GetStartupTimeout => TimeSpan.FromSeconds(240);
```
Comment (SbiCA) [inline TestContainers/Core/Containers/MsSqlContainer.cs:23]: Time out seem a bit high to me, but not sure what would be a good fit.

**Review by SbiCA — COMMENTED** [review]
I just had a look at it :)

**Pull request closed without merging**